src-cpp/rdkafkacpp_int.h

```diff
@@ -391,142 +391,98 @@ class ConfImpl : public Conf {
   }
 
   Conf::ConfResult get(const std::string &name, std::string &value) const {
-	  if (name.compare("dr_cb") == 0 ||
-	      name.compare("event_cb") == 0 ||
-	      name.compare("partitioner_cb") == 0 ||
-	      name.compare("partitioner_key_pointer_cb") == 0 ||
-	      name.compare("socket_cb") == 0 ||
-	      name.compare("open_cb") == 0 ||
-	      name.compare("rebalance_cb") == 0 ||
-	      name.compare("offset_commit_cb") == 0 ) {
-	      return Conf::CONF_INVALID;
-	  }
-	  size_t size;
-	  rd_kafka_conf_res_t res = RD_KAFKA_CONF_OK;
-	  if (rk_conf_) {
-		  if ((res = rd_kafka_conf_get(rk_conf_,
-			  name.c_str(), NULL, &size)) != RD_KAFKA_CONF_OK)
-			  return static_cast<Conf::ConfResult>(res);
-
-		  char valueCharArray[1000];
-		  res = rd_kafka_conf_get(rk_conf_, name.c_str(),
-				      (char *) valueCharArray, &size);
-		  value.assign(valueCharArray);
-		  if (res != RD_KAFKA_CONF_OK) {
-		      return static_cast<Conf::ConfResult>(res);
-		  }
-
-	  }
-	  else if (rkt_conf_) {
-		  if ((res = rd_kafka_topic_conf_get(rkt_conf_,
-			  name.c_str(), NULL, &size)) != RD_KAFKA_CONF_OK)
-			  return static_cast<Conf::ConfResult>(res);
-
-		  char valueCharArray[1000];
-		  res = rd_kafka_topic_conf_get(rkt_conf_, name.c_str(),
-					    (char *) valueCharArray, &size);
-		  value.assign(valueCharArray);
-		  if (res != RD_KAFKA_CONF_OK)
-			  return static_cast<Conf::ConfResult>(res);
-	  }
-
-	  return Conf::CONF_OK;
-  }
-
-  Conf::ConfResult get(const std::string &name, DeliveryReportCb *&dr_cb) const {
-      if (name != "dr_cb") {
+      if (name.compare("dr_cb") == 0 ||
+	  name.compare("event_cb") == 0 ||
+	  name.compare("partitioner_cb") == 0 ||
+	  name.compare("partitioner_key_pointer_cb") == 0 ||
+	  name.compare("socket_cb") == 0 ||
+	  name.compare("open_cb") == 0 ||
+	  name.compare("rebalance_cb") == 0 ||
+	  name.compare("offset_commit_cb") == 0 ) {
 	  return Conf::CONF_INVALID;
       }
-
-      if (!rk_conf_) {
-	  return Conf::CONF_INVALID;
+      size_t size;
+      rd_kafka_conf_res_t res = RD_KAFKA_CONF_OK;
+      if (rk_conf_) {
+	  if ((res = rd_kafka_conf_get(rk_conf_,
+				       name.c_str(), NULL, &size)) != RD_KAFKA_CONF_OK)
+	      return static_cast<Conf::ConfResult>(res);
+
+	  value.resize(size);
+	  if ((res = rd_kafka_conf_get(rk_conf_, name.c_str(),
+				       (char *)value.c_str(), &size)) != RD_KAFKA_CONF_OK)
+	      return static_cast<Conf::ConfResult>(res);
       }
-      dr_cb = this->dr_cb_;
+      else if (rkt_conf_) {
+	  if ((res = rd_kafka_topic_conf_get(rkt_conf_,
+					     name.c_str(), NULL, &size)) != RD_KAFKA_CONF_OK)
+	      return static_cast<Conf::ConfResult>(res);
+
+	  value.resize(size);
+	  if ((res = rd_kafka_topic_conf_get(rkt_conf_, name.c_str(),
+					     (char *)value.c_str(), &size)) != RD_KAFKA_CONF_OK)
+	      return static_cast<Conf::ConfResult>(res);
+      }
+
+      value.assign(value.c_str());
```

Comment:
well,
if done with stack based memory the Appveyor test fails if the Char[] is not initialized with a constant ( that's why I had this constructed with [1000] before.

I didn't want neither instantiate dynamically because, if something inside **rd_kafka_topic_conf_get** would throw and get caught outside of of Conf::get(string,string) that would make a leak.

but now that I think about it again, **rd_kafka_topic_conf_get** is pure C so it will never throw so I can instantiate a char array dynamically.

I'll do that.